From cf3b0d102855a4a90db3fb65a99e0009fcdd3755 Mon Sep 17 00:00:00 2001
From: Michael Pyne <mpyne@kde.org>
Date: Sat, 13 Oct 2018 19:09:50 -0400
Subject: kio_help: Fix crash in QCoreApplication when accessing help://.

A recent commit d428fc8e6447ede81f1e1911d0b66b39265672f3 removed old
custom crash handling code in favor of KCrash. But this actually adds a
dependency on there being a valid QCoreApplication object.

I took from the example of the similar kio_file to simply create the
QCoreApplication here.

BUG:399709

Differential Revision: https://phabricator.kde.org/D16189
---
 src/ioslaves/help/main.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ioslaves/help/main.cpp b/src/ioslaves/help/main.cpp
index f09987a..f800a9d 100644
--- src/ioslaves/help/main.cpp
+++ src/ioslaves/help/main.cpp
@@ -10,6 +10,7 @@ extern "C" int xmlLoadExtDtdDefaultValue;
 
 #include <QDebug>
 
+#include <QCoreApplication>
 #include <QString>
 
 #include <stdlib.h>
@@ -40,6 +41,9 @@ extern "C"
 {
     Q_DECL_EXPORT int kdemain(int argc, char **argv)
     {
+        QCoreApplication app(argc, argv);   // needed for KCrash
+        app.setApplicationName(QStringLiteral("kio_help"));
+
         KDocTools::setupStandardDirs();
 
         //qDebug() << "Starting " << getpid();
-- 
cgit v0.11.2

